[TNA][Review] Review of the New Architecture guide

cipolleschi opened the pull request:
This Pr is a first review of the New Architecture guide.

The main focus is to review the Playbook having in mind the release of 0.70.0.

It also tries to rename all the `new architecture` to `New Architecture` and all the `CodeGen` and `code-gen` to `Codegen`.

## TODO

- [x] review Enabling TurboModule on Android with Autolinking
- [x] review Enabling Fabric Components on Android with Autolinking
- [x] add banners on Native Modules and Native Components

cipolleschi commented on docs/new-architecture-app-intro.md:
````diff
@@ -7,40 +7,25 @@ import NewArchitectureWarning from './\_markdown-new-architecture-warning.mdx';
+yarn add react-native@0.0.0-20220705-2052-71e89213a
 ```
 
-## Install react-native-codegen
````
`yarn add react-native@xxxx` should install the package as well.

cortinico replied:
Yup that's correct 👍 This was old

cipolleschi commented on docs/new-architecture-app-intro.md:
```diff
@@ -54,18 +39,6 @@ cd android && ./gradlew wrapper --gradle-version 7.3 --distribution-type=all
 
 While the AGP version should be updated inside the **top level** `build.gradle` file at the `com.android.tools.build:gradle` dependency line.
 
-If you’re set with it, let’s now install the new Gradle plugin which is distributed through a NPM package called [**`react-native-gradle-plugin`**](https://www.npmjs.com/package/react-native-gradle-plugin). You can do so with:
```
`yarn add react-native@xxxx` should install the package as well.

cipolleschi commented on docs/new-architecture-app-modules-android.md:
```diff
@@ -36,10 +36,15 @@ android {
                 cppFlags "-std=c++17"
                 targets "myapplication_appmodules"
+                // Fix for windows limit on number of character in file paths and in command lines
+                if (Os.isFamily(Os.FAMILY_WINDOWS)) {
```
I tried to replicate the template here. I think that this is helpful if people develops using windows.

cipolleschi commented on docs/new-architecture-appendix.md:
```diff
@@ -27,11 +27,31 @@ You may use the following table as a reference for which types are supported and
+
+You may use the following table as a reference for which types are supported and what they map to in each platform:
+
+| TypeScript Type                                | Nullable Support?                                        | Android (Java)                       | iOS                                                            | Note                                                                           |
```
I'm not sure whether this table is correct or not. Who has a good knowledge of TypeScript to check this?

cipolleschi commented on docs/new-architecture-library-android.md:
```diff
@@ -7,41 +7,9 @@ import NewArchitectureWarning from './\_markdown-new-architecture-warning.mdx';
-Once you have defined the JavaScript specs for your native modules as part of the [prerequisites](new-architecture-library-intro) and followed the Android/Gradle setup, you are now ready to migrate your library to the new architecture. Here are the steps you can follow to accomplish this.
+Once you have defined the JavaScript specs for your native modules as part of the [prerequisites](new-architecture-library-intro), setup the configuration of the Codegen, and followed the Android/Gradle setup, you are now ready to migrate your library to the new architecture. Here are the steps you can follow to accomplish this.
 
-### 1. Configure Codegen in your Gradle File
```
Moved to the library intro, now it is shared with iOS

cipolleschi commented on docs/the-new-architecture/_markdown_native_deprecation.mdx:
```diff
@@ -0,0 +1,4 @@
+:::caution
+Native Module and Native Components are legacy technologies bound to the legacy architecture and so, deprecated.
+The New Architecture uses [TurboModule](pillars-turbomodules) and [Fabric Components](pillars-fabric-components) to achieve similar results.
```
Wasn't the idea to remove those banners with 0.70.0? 🤔

cortinico replied:
> Wasn't the idea to remove those banners with 0.70.0? 🤔

Nope we should keep those banners as long as we keep on calling the New Architecture experimental. We can reword them as we wish.

cipolleschi commented:
@cortinico I reviewed the PR, updated the playbook and added some other fixes.

There are a couple of things left to discuss:
- [ ] Codegen on Android is not working correctly with the new setup
- [ ] What do we want to do with banners and the structure of the website

cortinico commented:
> * [ ]  Codegen on Android is not working correctly with the new setup

Addressed in #34298

> * [ ]  What do we want to do with banners and the structure of the website

Let's discuss about this offline? 👍

arati-1 reviewed (commented):
Hi @cipolleschi I have given minor comments. Take a look.

cipolleschi commented:
Thanks @arati-1, I applied them to the container PR.
